internal/k8s/controller.go

```diff
@@ -2530,6 +2530,14 @@ func (lbc *LoadBalancerController) createVirtualServerEx(virtualServer *conf_v1.
 			if err != nil {
 				glog.Warningf("Error getting WAF policies for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
 			}
+
+			if sr.Dos != "" {
```

Comment:
currently this looks like a bug (I haven't tested it by running the IC in the cluster). 
what will happen if the DOS is configured for NGINX OSS IC or NGINX Plus without DOS enabled? this is related to this comment https://github.com/nginxinc/kubernetes-ingress/pull/2160#discussion_r753627593

Reply:
fixed